conda/core/index.py

```diff
@@ -88,7 +88,8 @@ def get_index(channel_urls=(), prepend=True, platform=None,
     if context.offline and unknown is None:
         unknown = True
 
-    channel_priority_map = prioritize_channels(channel_urls, platform=platform)
+    subdirs = (platform, 'noarch') if platform is not None else context.subdirs
+    channel_priority_map = prioritize_channels(channel_urls, subdirs=subdirs)
```

Comment:
This is the `get_index()` function.  The API remains the same right now.  You can see I'm translating the `platform` argument into `subdirs`.  At some point, we could switch out a `subdirs` for `platform` if we wanted.